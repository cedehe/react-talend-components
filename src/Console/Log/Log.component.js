import React from 'react';
import classnames from 'classnames';
import Actions from '../../Actions';
import theme from './Log.scss';

/**
 * @param {object} props react props
 * @example
<Log name="Hello world"></Log>
 */
function Log(props) {
	const wrapper = classnames(
		'tc-console-log',
		theme.wrapper
	);
	const msg = classnames(
		'tc-console-log-msg',
		`tc-console-log-msg-${props.level}`,
		theme[props.level],
		theme.msg
	);
	return (
		<div className={wrapper}>
			<p className={msg}>
				{props.msg}
				{props.actions ? (<Actions actions={props.actions} />) : null}
			</p>
		</div>
	);
}

Log.propTypes = {
	msg: React.PropTypes.string,
	level: React.PropTypes.oneOf(
		['info', 'error', 'warning']
	),
	actions: React.PropTypes.arrayOf(
		React.PropTypes.object
	),
};

export default Log;
