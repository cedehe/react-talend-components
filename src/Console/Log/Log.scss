// Log styles

$tc-console-log-msg-info: #3C763D !default;
$tc-console-log-msg-warning: #8A6D3B !default;
$tc-console-log-msg-error: #A94442 !default;

.wrapper {
	width: 100%;
}

.msg {
	margin: 0;
}

.info {
	color: $tc-console-log-msg-info;
}

.warning {
	color: $tc-console-log-msg-warning;
}

.error {
	color: $tc-console-log-msg-error;
}
