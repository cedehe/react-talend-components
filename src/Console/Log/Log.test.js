import React from 'react';
import renderer from 'react-test-renderer';

import Log from './Log.component';

jest.mock('react-dom');

const info = { msg: 'Hello world, how are you ?' };
const error = { msg: 'Something goes wrong here', level: 'error' };
const warning = {
	msg: 'Something goes wrong but can be fixed',
	level: 'warning',
	actions: [
		{
			label: 'fixme',
			onClick: jest.fn(),
			icon: 'fa fa-bug',
		},
	],
};

describe('Log', () => {
	it('should render a log info by default', () => {
		const wrapper = renderer.create(
			<Log {...info} />).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
	it('should render a log warning', () => {
		const wrapper = renderer.create(
			<Log {...warning} />).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
	it('should render an error', () => {
		const wrapper = renderer.create(
			<Log {...error} />).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
});
