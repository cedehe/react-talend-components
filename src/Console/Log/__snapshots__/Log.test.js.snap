exports[`Log should render a log info by default 1`] = `
<div
  className="tc-console-log">
  <p
    className="tc-console-log-msg tc-console-log-msg-undefined">
    Hello world, how are you ?
  </p>
</div>
`;

exports[`Log should render a log warning 1`] = `
<div
  className="tc-console-log">
  <p
    className="tc-console-log-msg tc-console-log-msg-warning">
    Something goes wrong but can be fixed
    <div
      className="tc-actions btn-group">
      <button
        className="btn btn-default"
        disabled={false}
        onClick={[Function]}
        role={null}
        type="button">
        <i
          className="fa fa-bug" />
        <span>
          fixme
        </span>
      </button>
    </div>
  </p>
</div>
`;

exports[`Log should render an error 1`] = `
<div
  className="tc-console-log">
  <p
    className="tc-console-log-msg tc-console-log-msg-error">
    Something goes wrong here
  </p>
</div>
`;
